File: editor/en/[1137]N-th Tribonacci Number.py
# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def tribonacci(self, n: int) -> int:
        if n == 0:
            return 0
        elif n == 1 or n == 2:
            return 1

        t0, t1, t2 = 0, 1, 1

        for i in range(3, n + 1):
            current = t0 + t1 + t2
            t0, t1, t2 = t1, t2, current

        return current
# leetcode submit region end(Prohibit modification and deletion)

File: editor/en/[1137]N-th Tribonacci Number.java
//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public int tribonacci(int n) {
        if (n <= 2) {
            return n == 0 ? 0 : 1;
        }

        int t0 = 0;
        int t1 = 1;
        int t2 = 1;
        int curr = 0;

        for (int i = 3; i <= n; i++) {
            curr = t0 + t1 + t2;
            t0 = t1;
            t1 = t2;
            t2 = curr;
        }

        return curr;
    }
}
//leetcode submit region end(Prohibit modification and deletion)
